upstairs/src/volume.rs

```diff
@@ -12,8 +12,7 @@ pub struct Volume {
     uuid: Uuid,
 
     sub_volumes: Vec<SubVolume>,
-    read_only_parent: Option<SubVolume>, /* TODO mutex so migration task can
-                                          * remove */
+    read_only_parent: Mutex<Option<SubVolume>>,
```

Comment:
Actually, it's also possible that we may not need this to be a Mutex, or even modifiable - the scrubber could run, signal to Nexus that it's done, then Nexus could PUT a new VolumeConstructionRequest without the read-only parent to Propolis, causing the Volume to be reinitialized without the read-only parent. That way all modification is Nexus driven, and there's no drift between what VolumeConstructionRequests are in the DB vs what are live.

Reply:
I like both the RW lock and the no Mutex at all ideas.

I think I'm going to pull the `read_only_parent: Mutex<Option<SubVolume>>>` change out of this PR and just push (after addressing comments) the scrubber part and the tests.

We can determine next steps with RW lock, Pushing it to Nexus, or some other solution.
I also want to implement the scrubber high water mark, so the volume layer can tell where the scrubber has been and reads under that point can go directly to the SubVolume.